### ISSUE TITLE
XmlDriver incorrectly parses boolean option values

### ISSUE THREAD

**Issue by kalimatas** [doctrine/doctrine2#6129]
I have a mapping for a column like that:

```
<field name="alpha2Code" column="country_code" type="string" length="2" column-definition="CHAR(2) NOT NULL">
    <options>
        <option name="fixed">false</option>
    </options>
</field>
```

Despite that I specified `false` as a value of `fixed` option, it will be `true`, because in `XmlDriver::_parseOptions` method the string `false` will be casted to `true` with `(bool)`:

```
if ($option->count()) {
    $value = $this->_parseOptions($option->children());
} else {
    // here
    $value = (string) $option;
}
```

It probably should be:

```
$value = $this->evaluateBoolean((string) $option);
```

**Comment by Ocramius** [comment]
@kalimatas can this be converted into a test case?

**Comment by kalimatas** [comment]
I'm on it right now.

**Comment by Ocramius** [comment]
Tests in #6130

**Comment by kalimatas** [comment]
I would like also to fix it, though I'm not sure, if maintaining an array of option names, that are supposed to be boolean in the `_parseOptions()` method , is a nice solution.

**Comment by Ocramius** [comment]
@kalimatas give it a shot - it can be a quick fix with a later refactoring, but if there's a possible BC break, we need to expose that as well.

**Comment by kalimatas** [comment]
I created a generic test for all drivers.

**Comment by Ocramius** [comment]
Closing as per work done in #6130

**Comment by Ocramius** [comment]
Note: this is only merged into `2.6.0` due to risk of regressions in the mapping driver change.
